Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -446,7 +446,7 @@ class Connection : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<Connec
     template<typename T> using SendSyncResult = ConnectionSendSyncResult<T>;
     template<typename T> SendSyncResult<T> sendSync(T&& message, uint64_t destinationID, Timeout = Timeout::infinity(), OptionSet<SendSyncOption> sendSyncOptions = { }); // Main thread only.
 
-    template<typename> Error waitForAndDispatchImmediately(uint64_t destinationID, Timeout, OptionSet<WaitForOption> waitForOptions = { }); // Main thread only.
+    template<typename... Ts> Error waitForAndDispatchImmediately(uint64_t destinationID, Timeout, OptionSet<WaitForOption> waitForOptions = { }); // Main thread only.
```

Comment:
Would you be willing to omit this change? It's fairly niche use-case, and it brings the burden that we need to support it in both here and stream. And we should test that the implementation is actually correct.

Could we instead use a single message?
E.g.

Reply:
I can, if you'd prefer.

I'll have to combine all three (WillCommitLayerTree, CommitLayerTree, CommitLayerTreeNotTriggered) into a single 'update layer tree state' message. Bit clunky, but happy to have the complexity there rather than IPC.